numpy/core/_dtype.py

```diff
@@ -201,14 +201,8 @@ def _datetime_metadata_str(dtype):
 def _struct_dict_str(dtype, includealignedflag):
     # unpack the fields dictionary into ls
     names = dtype.names
-    fld_dtypes = []
-    offsets = []
-    titles = []
-    for name in names:
-        fld_dtype, offset, title = _unpack_field(*dtype.fields[name])
-        fld_dtypes.append(fld_dtype)
-        offsets.append(offset)
-        titles.append(title)
+    fld_dtypes, offsets, titles = zip(
+            *[_unpack_field(*dtype.fields[name]) for name in names])
```

Comment:
This is actually different, and produces tuples instead of lists. I'm not sure if it matters, but it's worth noting that it is different.